### PR TITLE
More Bits: More small improvements on the way to dApp signing

### PR DESCRIPTION
**Pull request by Shadowfiend**
Commits describe it all here.

See #366 .

### CODE REVIEW

File: background/main.ts [hunk 1]
```diff
@@ -315,43 +315,36 @@ export default class Main extends BaseService<never> {
...
-          nonce: resolvedNonce,
-        }
-
-        transaction.gasLimit = await this.chainService.estimateGasLimit(
-          getEthereumNetwork(),
-          transaction
+    transactionConstructionSliceEmitter.on("updateOptions", async (options) => {
+      // TODO Deal with pending transactions.
+      const resolvedNonce =
+        await this.chainService.pollingProviders.ethereum.getTransactionCount(
```
Comment (mhluongo) [inline background/main.ts:321]: I expect we'll be revisiting this choice down the line :)

File: background/main.ts [hunk 2]
```diff
@@ -337,11 +337,26 @@ export default class Main extends BaseService<never> {
...
-        transaction
-      )
-      this.store.dispatch(transactionRequest(transaction))
+      try {
+        transaction.gasLimit = await this.chainService.estimateGasLimit(
+          getEthereumNetwork(),
+          transaction
+        )
+
+        this.store.dispatch(
```
Comment (mhluongo) [inline background/main.ts:346]: It's the little things

**Review by mhluongo — APPROVED** [review]
Looking good.

File: background/main.ts (hunk 1 above)
Comment (Shadowfiend) [reply background/main.ts:321]: So much revisiting. This whole code chunk really needs to push into the chain service.

**Pull request merged**